### DISCUSSION TITLE
How do I reset my Supabase database password?

### DISCUSSION THREAD

**Discussion by KevinBrolly** [supabase/supabase#20929] [Troubleshooting]
_This is a copy of a troubleshooting article on Supabase's docs site. It may be missing some details from the original. View the [original article](https://supabase.com/docs/guides/troubleshooting/how-do-i-reset-my-supabase-database-password-oTs5sB)._

You can reset your database password from the [**Database > Settings**](https://supabase.com/dashboard/project/_/database/settings) section of the Dashboard.

If connections through the [Shared Pooler (Supavisor)](https://supabase.com/docs/guides/database/connecting-to-postgres) start failing right after you reset it — including for the built-in `postgres` role — this `28P01` error is expected, and retrying with the new password usually succeeds. See [Supavisor error: "password authentication failed" after rotating database password](https://supabase.com/docs/guides/troubleshooting/supavisor-error-password-authentication-failed-after-password-rotation) for more detail.

**Comment by fayazkhan121** [comment]
Log in to your Supabase account: Go to [Supabase](https://supabase.com/) and log in.
Open your project: Click on the project where you want to reset the database password.
Go to the Database settings:
Navigate to the Settings tab in the left-hand menu.
Select Database.
Reset the password:
Look for the "Connection String" or "Database Configuration" section.
Click on the "Edit Password" button (if available) or the equivalent option.
Enter a new password and save the changes.
Update connection strings:
After resetting the password, update all applications and services that connect to this database with the new password.

**Comment by jwansoo** [comment]
I follow your command but error occur like this I don.t what can I do  please let me know
I use vuejs and vs code
 supabase db reset --linked
s/database
Enter your database password:
Do you want to reset the remote database? [y/N] y
Resetting remote database...
failed to connect to postgres: failed to connect to `host=aws-0-ap-southeast-1.pooler.supabase.com user=postgres.jkkdoxuajqloldoijbxm database=postgres`: server error (FATAL: Address not in tenant allow_list: {218, 150, 134, 176} (SQLSTATE XX000))
Try rerunning the command with --debug to troubleshoot the error.

**Comment by Monty1122** [comment]
hey folks i really cant find where to reset the database password - the docs seem out of date - for example when i go to supabase and then Database i see; Database Management, Access Control,Platform,Tools
but no Database Settings - there is a main Settings fro the project but no way to see or update the database password there - anybody got any other suggestions?

**Reply by GaryAustin1** [reply]
@Monty1122 Go to Project Settings tab.  Then Database.

**Comment by Byaxy** [comment]
Log into your supabase account, and navigate to the project where you want to change the password. In the bottom left, just above your profile image, click on "Project Settings". On the left sidebar menu, under the "Configuration" section, click on "Database". This should display the "Database Settings" under which you will see a "Database password" section with a "Reset database password" button on the right.

**Comment by daryllxd** [comment]
Like this: 
![image](https://github.com/user-attachments/assets/1d96d465-40fa-4f3a-8b10-059440695f97)

**Comment by Byaxy** [comment]
Yes...
That will open a dialog where you can enter the new password.

**Comment by zengqingfu1442** [comment]
I cannot reset the database password. Network error? but its always the same error.
![image](https://github.com/user-attachments/assets/67a6e0b0-3a3b-4b8e-bc49-f53e83195c63)

**Reply by e3d** [reply]
I got the same error - Failed to reset password.

**Reply by AprilPolubiec** [reply]
I have been getting this as well - any updates?

**Reply by raushankumar90** [reply]
Facing Same Issue

**Reply by ghost** [reply]
This problem occurs using local supabase too... How to fix this?

**Reply by spjhon** [reply]
any news?, I have the same error, I tried everything.

**Comment by spjhon** [comment]
I HAVE THE SAME ERROR

**Comment by aantti** [comment]
A bit hard to understand what exact problems are referred above by various folks 🙈 It is possible to change the database password in managed Supabase platform - check Database → Settings.

However, it is _not_ possible to change the password in the UI for CLI/local-dev and self-hosted Supabase. For [self-hosted](https://supabase.com/docs/guides/self-hosting), see here - https://supabase.com/docs/guides/self-hosting/docker#changing-database-password

If anyone is experiencing difficulties with resetting database password in [managed Supabase](https://supabase.com/dashboard) - best way to resolve would be via a support ticket.

**Comment by uday770202** [comment]
Hi @aantti, I am running Supabase locally using npx supabase start, but I’m unable to figure out how to change the database password. Is there a way to update the DB password in a local Supabase setup?

**Reply by aantti** [reply]
I don't think it's going to be easy, @uday770202 :(

For one, it's defined here:

https://github.com/supabase/cli/blob/b3449760723eddbe32281f804dd13212a57b8062/pkg/config/config.go#L344

and then it's populated in the env-vars across all containers that are being started by CLI (you can check via `docker inspect`).

If you need a unique password, I guess the easiest would be to clone the CLI repo, change the default password in the code, and rebuild for local use (via `go build` & `go install`).

If you need to change the password for the existing setup, it's probably possible to use something like we have for self-hosted:

https://github.com/supabase/supabase/blob/7013388c0c6204061948a9d2129289031d03fb59/docker/utils/db-passwd.sh#L111

You could connect to the Postgres container and run the same. Still, CLI would use the hardcoded password when starting the containers.

Bear in mind, [CLI](https://supabase.com/docs/guides/local-development) is **not meant to be used in a production or open dev/staging/test environment** - it should never be exposed to any public traffic. A lot of credentials in CLI are default values and hardcoded. CLI is supposed to be used in an local, isolated, tightly controlled setup only.
